File: backend/Models/Album.cs

using backend.Enums;
namespace backend.Models;

public class Album
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public int ArtistId { get; set; }
    public int? ReleaseYear { get; set; }
    public string? CoverImageUrl { get; set; }
    public int Rating { get; set; }
    public int ReviewCount { get; set; }
    public Genre? Genre { get; set; }    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/interfaces/Album.ts
export interface IAlbum {
  id: number;
  title: string;
  artistId: number;
  releaseYear?: number | null;
  coverImageUrl?: string | null;
  rating: number;
  reviewCount: number;
  genre?: number | null;
  createdAt: string;
  artist?: {
    id: number;
    name: string;
  } | null;
}